File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.List;
class ResistorColorTrio {
    final List<String> colorsInOrder = List.of("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");
    String label(String[] colors) {
        long result = (10 * colorsInOrder.indexOf(colors[0]) + colorsInOrder.indexOf(colors[1])) * (long) Math.pow(10, colorsInOrder.indexOf(colors[2]));
        return result < 1000 ? result + " ohms" : result < 1000000 ? result / 1000 + " kiloohms" : result < 1000000000 ? result / 1000000 + " megaohms" : result / 1000000000 + " gigaohms";
    }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
def label(colors: list[str]) -> str:
    COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
    UNITS = ["", "kilo", "mega", "giga"]
    result = (COLORS.index(colors[0])*10 + COLORS.index(colors[1])) * 10**COLORS.index(colors[2])
    unit = 0
    while result != 0 and result % 1000 == 0 and unit < len(UNITS) - 1:
        result //= 1000
        unit += 1
    return str(result) + " " + UNITS[unit] + "ohms"
